src/widgets/infinite-hits/infinite-hits.tsx

```diff
@@ -12,20 +14,58 @@ import {
 } from '../../lib/utils';
 import { component } from '../../lib/suit';
 import { withInsights, withInsightsListener } from '../../lib/insights';
+import {
+  WidgetFactory,
+  Template,
+  Hit,
+  InsightsClientWrapper,
+} from '../../types';
 
 const withUsage = createDocumentationMessageGenerator({
   name: 'infinite-hits',
 });
 const suit = component('InfiniteHits');
 const InfiniteHitsWithInsightsListener = withInsightsListener(InfiniteHits);
 
+export type InfiniteHitsCSSClasses = {
+  root: string | string[];
+  emptyRoot: string | string[];
+  item: string | string[];
+  list: string | string[];
+  loadPrevious: string | string[];
+  disabledLoadPrevious: string | string[];
+  loadMore: string | string[];
+  disabledLoadMore: string | string[];
+};
+
+export type InfiniteHitsTemplates = {
+  empty: Template<void>;
+  showPreviousText: Template<void>;
+  showMoreText: Template<void>;
+  item: Template<Hit>;
+};
+
+export type InfiniteHitsRendererWidgetParams = {
+  escapeHTML: boolean;
+  transformItems: (items: any[]) => any;
```

Comment:
I can't find any reasons for the return type not to be an array for this widget.

Reply:
I don't know either, but it was typed that way in https://github.com/algolia/instantsearch.js/blob/9dca548/src/connectors/query-rules/connectQueryRules.ts#L29.

Can be good to be stricter on that if we agree that we can only return an Array here.

Reply:
For Query Rules it was a bit different. We wanted to allow people to use `transformItems` to process the `items` as a single one because most of the times, this is what you want. I'm not sure we should recommend this pattern though.

Reply:
Oh, ok. I did not saw that the context was different for the two widgets. I updated the definition to assert that we always return an Array.